pipelines/build/common/kick_off_build.groovy

```diff
@@ -14,15 +17,68 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Downstream job root executor file, it sets up the library and runs the bash script setup job. 
-def builder;
+/*
+Downstream job root executor file, it sets up the library and runs the bash script setup job.
+*/
+
+// We have to declare JSON defaults again because we're utilising it's content at start of job
+def LOCAL_DEFAULTS_JSON = new JsonSlurper().parseText(DEFAULTS_JSON) as Map
+if (!LOCAL_DEFAULTS_JSON) {
+    throw new Exception("[ERROR] No User Defaults JSON found! Please ensure the DEFAULTS_JSON parameter is populated and not altered during parameter declaration.")
+}
+def ADOPT_DEFAULTS_JSON = new JsonSlurper().parseText(ADOPT_DEFAULTS_JSON) as Map
+if (!ADOPT_DEFAULTS_JSON) {
+    throw new Exception("[ERROR] No Adopt Defaults JSON found! Please ensure the ADOPT_DEFAULTS_JSON parameter is populated and not altered during parameter declaration.")
+}
+
+def libraryPath = (params.CUSTOM_LIBRARY_LOCATION) ?: LOCAL_DEFAULTS_JSON["importLibraryScript"]
+def baseFilePath = (params.CUSTOM_BASEFILE_LOCATION) ?: LOCAL_DEFAULTS_JSON["baseFileDirectories"]["downstream"]
+
+def userRemoteConfigs = [:]
+def downstreamBuilder = null
 node("master") {
+    /*
```

Comment:
Not sure if any of this can be extracted to functions given how Jenkins DSL works?

Reply:
The [try-catch block](https://github.com/AdoptOpenJDK/openjdk-build/pull/2132/files/17bbc5bc61119a6e4b08afb8ec73522b09e57294#diff-a385c8842d53dff1a13532ec0e118480d4ebce5fbca4b3b3ad469e5d57f0f1eaR56-R72) probably could? I don't see what else could be, the file is already pretty compacted

Reply:
No biggie, give it a go and see :-)